encoding/descaling.md

```diff
@@ -0,0 +1,356 @@
+This is because the best available sources to consumers,
+usually Blu-rays,
+aren't lossless.
+There's always going to be some differences
+from the original masters
+which makes it impossible
+to perfectly descale something.
+However, usually those differences
+are so small that they're negligible.
+If you run into a case where
+you can only get high relative errors,
+descaling can be highly destructive.
+It's instead recommended to downscale
+as you would normally,
+or to not mess with scaling at all.
+
+[getnative]: https://github.com/Infiziert90/getnative
+[descale]: https://github.com/BluBb-mADe/vapoursynth-descale
+
+
+## Finding out the native resolution
+
+To figure out what
+the native resolution of an anime is,
+first you need a good frame to test.
+Ideally you'll want a bright frame
+with as little blur as possible of high quality
+(Blu-ray or very good webstreams).
+It also helps to not have
+too many post-processed elements in the picture.
+Whilst it is most definitely possible
+to get pretty good results with "bad" frames,
+you'll want to generally lower that chance
+as much as possible.
+
+Here's some examples of "bad" frames.
+
+![Manaria Friends Ep01 frame 1](images/descale_manaria01.png)
+
+This picture is dark.
+It also has some effects over it.
+
+![Manaria Friends Ep01 frame 2](images/descale_manaria02.png)
+
+This picture is also very dark,
+and has even more effects over it.
+
+![Manaria Friends Ep01 frame 3](images/descale_manaria03.png)
+
+Heavy dynamic grain will almost always give bad results.
+
+![Manaria Friends Ep01 frame 4](images/descale_manaria04.png)
+
+This is a nice frame to use as reference.
+The background is a bit blurry,
+but it isn't full of effects
+and is fairly bright.
+The lineart is very clear.
+
+We will now make use of the getnative.py script
+to figure out what resolution
+this anime was produced at.
+Run the following in your terminal:
+
+```bash
+$ python getnative.py "Manaria Friends Ep01 frame 4.png"
+```
+
+It should show the following:
+
+```
+Using imwri as source filter
+501/501
+Kernel: bicubic AR: 1.78 B: 0.33 C: 0.33
+Native resolution(s) (best guess): 878p
+done in 18.39s
+```
+
+If you look in the directory
+that you ran the script in,
+you will find a new folder
+called "getnative".
+You can find the following graph
+in there as well:
+
+![Manaria_Friends-ep1_04_graph](images/descale_graph.png)
+
+You should be looking for a low relative error.
+In this case it very clearly points to 878p.
+
+
+## Descaling
+
+Now it's time to actually start descaling.
+Open up your Vapoursynth editor of choice,
+and import the clip:
+
+```Py
+src = lvf.src(r"BDMV/[BDMV][190302][マナリアフレンズ I]/BD/BDMV/STREAM/00007.m2ts")
+```
+
+The next issue is figuring out
+what was used to [upsample](resampling.md#upsampling) the show.
+By default,
+getnative.py checks with Mitchell-Netravali
+(bicubic b=1/3, c=1/3).
+However, it might have also been upscaled
+using other kernels,
+like Spline or Bilinear.
```

Comment:
An appendix or a small section about popular kernels would be useful for beginners.